front/src/applications/operationalStudies/hooks/useScenarioData.ts

```
@@ -155,13 +156,15 @@ const useScenarioData = (scenario: ScenarioResponse, infra: InfraWithState) => {
     [projectedTrainsById]
   );
 
-  const timetableItemUsedForProjection = useMemo(
-    () =>
-      trainIdUsedForProjection && !isOccurrenceId(trainIdUsedForProjection)
-        ? timetableItemsById.get(trainIdUsedForProjection)
-        : undefined,
-    [trainIdUsedForProjection, timetableItems]
-  );
+  const timetableItemUsedForProjection = useMemo(() => {
+    if (!trainIdUsedForProjection) return undefined;
+
+    const timetableItemId = isOccurrenceId(trainIdUsedForProjection)
+      ? extractPacedTrainIdFromOccurrenceId(trainIdUsedForProjection)
```

Hm, here we grab the base paced train and ignore exceptions. This means any changes made by the exception are not reflected in `timetableItemUsedForProjection`. What if an exception has a different path, for instance?

Among other things, in `SimulationResults` we call `useGetProjectedTrainOperationalPoints()` with `projectionData?.timetableItem`. Wouldn't this result in the wrong OP list being computed?

---

Hum yes you're right, this is very confusing...

---

Done